homeassistant/components/yardian/__init__.py

```diff
@@ -2,17 +2,46 @@
 from __future__ import annotations
 
 from pyyardian import AsyncYardianClient
+import voluptuous as vol
 
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_ACCESS_TOKEN, CONF_HOST, Platform
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType
 
 from .const import DOMAIN
 from .coordinator import YardianUpdateCoordinator
 
 PLATFORMS: list[Platform] = [Platform.SWITCH]
 
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Optional("duration", default={}): vol.All(
+                    {
+                        vol.Required(cv.string): cv.positive_int,
+                    },
+                    vol.Length(min=1),
+                ),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up for Yardian."""
+
+    hass.data[DOMAIN] = {
+        "config": config.get(DOMAIN, {}),
+    }
+
+    return True
+
```

Comment:
Please remove the yaml configuration part as integration, which is communicating with a device/service, must be set up only via UI. Please see https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

Reply:
Oh.. I've just remembered that someone told me about this in my last PR. Sorry for did this again, I'll survey on the replacement and implement it in the future PR. For this time I'll just remove it.